spec/chromium-spec.js

```diff
@@ -1411,6 +1411,83 @@ describe('chromium feature', () => {
       await new Promise((resolve) => { utter.onend = resolve })
     })
   })
+
+  describe('focus handling', () => {
+    beforeEach(async () => {
+      w = new BrowserWindow({
+        show: true
+      })
+
+      await w.loadFile(path.join(fixtures, 'pages', 'tab-focus-loop-elements.html'))
+      w.focus()
+    })
+
+    const getCurrentFocusedElement = async () => {
+      const focusedElementIdResult = emittedOnce(ipcMain, 'focused-element-id')
+      w.webContents.send('get-focused-element-id')
+      const [, focusedElementId, focusedElementTag] = await focusedElementIdResult
+      return [focusedElementId, focusedElementTag]
+    }
+
+    const getFinalFocusedElementId = async () => {
+      const [focusedElementId, focusedElementTag] = await getCurrentFocusedElement()
+      if (focusedElementTag === 'BODY') {
+        // Wait for Electron to react and move the focus back
+        // to the first / last element if necessary.
+        const waitTimeMs = 10
```

Comment:
Hate this magic constant, but the test has to wait a bit until Electron reacts to Chromium's notification that the focus has gone through all the focusable elements and sets it back to the first or last element. Any idea to implement this in a better way?

Reply:
Rather than requesting the currently focused element, the loaded page can signal when an element receives focus and the test can validate with it.

Reply:
Excellent idea, changed accordingly.